### PR TITLE
Feature/get conversations

### PR DESCRIPTION
**Pull request by Rodrigo-Miranda-0**
#### Board:
* [Ticket #HyswrCKs](https://trello.com/c/HyswrCKs/10-3-as-a-user-i-should-be-able-to-see-all-my-conversations)
---
#### Description:

* Allows for a user to view all his conversations
---
#### Notes:
*
---
#### Tasks:
- [x] Get all conversations
---
#### Risk:
*
---
#### Preview:

### CODE REVIEW

File: app/views/api/v1/application_mailer/target_match.html.erb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<html>
+  <head>
+    <meta content='text/html; charset=UTF-8' http-equiv='Content-Type' />
+  </head>
+  <body>
+    <h1><%= @user.email %> your target <%= @target.title %> just got matched!</h1>
+    <p>
+      You can now have a conversation with the matched user.
```
Comment (lauperalti) [inline app/views/api/v1/application_mailer/target_match.html.erb]: Could we use Internationalization for the plain texts ?

**Review by lauperalti — APPROVED** [review]
👏

File: app/views/api/v1/application_mailer/target_match.html.erb (hunk 1 above)
Comment (juanmferreira93) [reply app/views/api/v1/application_mailer/target_match.html.erb]: Agree, check this gem: https://github.com/ruby-i18n/i18n

File: spec/requests/api/v1/conversations/index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+describe 'Get conversations', type: :request do
```
Comment (juanmferreira93) [inline spec/requests/api/v1/conversations/index_spec.rb]:
```suggestion
describe 'GET conversations', type: :request do
```

File: spec/requests/api/v1/conversations/index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  let(:user) { create(:user) }
+  let(:another_user) { create(:user) }
+  let!(:conversation) { create(:conversation, user1_id: user.id, user2_id: another_user.id) }
+  let(:headers) { user.create_new_auth_token }
+
+  context 'when user is logged in' do
+    it 'returns all user conversations' do
+      subject
+      p response.body
```
Comment (juanmferreira93) [inline spec/requests/api/v1/conversations/index_spec.rb]: 👀 remove this please

File: spec/requests/api/v1/targets/match_spec.rb [hunk 4]
```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
+
+describe 'Match Targets', type: :request do
```
Comment (juanmferreira93) [inline spec/requests/api/v1/targets/match_spec.rb]:
```suggestion
describe 'POST Targets', type: :request do
```
Here you are testing the POST endpoint for Target creation, then you will have a case for when there are matched targets and also for there not are matched targets (they should be context below)

File: spec/requests/api/v1/targets/match_spec.rb [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        title:,
+        radius:,
+        topic_id: topic.id
+      },
+      latitude: -63.59386082773612,
+      longitude: -5.199173507365742
+    }
+  end
+
+  context 'when success' do
```
Comment (juanmferreira93) [inline spec/requests/api/v1/targets/match_spec.rb]: here I suggest you to add new context below, one for when there are matched targets and other for when there are not matched targets.. so for the one that there are matched targets, you need to check if the Conversation where created and that.. for the other, ensure that no Conversation where created

**Review by juanmferreira93 — APPROVED** [review]

**Review by jpascual1994 — APPROVED** [review]

**Pull request merged**